verifiable/did_doc.go

```diff
@@ -126,12 +126,25 @@ type StateInfo struct {
 
 // GistInfo representation state of gist root.
 type GistInfo struct {
-	Root                string `json:"root"`
-	ReplacedByRoot      string `json:"replacedByRoot"`
-	CreatedAtTimestamp  string `json:"createdAtTimestamp"`
-	ReplacedAtTimestamp string `json:"replacedAtTimestamp"`
-	CreatedAtBlock      string `json:"createdAtBlock"`
-	ReplacedAtBlock     string `json:"replacedAtBlock"`
+	Root                string     `json:"root"`
+	ReplacedByRoot      string     `json:"replacedByRoot"`
+	CreatedAtTimestamp  string     `json:"createdAtTimestamp"`
+	ReplacedAtTimestamp string     `json:"replacedAtTimestamp"`
+	CreatedAtBlock      string     `json:"createdAtBlock"`
+	ReplacedAtBlock     string     `json:"replacedAtBlock"`
+	Proof               *GistProof `json:"proof"`
+}
+
+// GistProof representation proof of gist root.
+type GistProof struct {
+	Root         string   `json:"root"`
+	Existence    bool     `json:"existence"`
+	Siblings     []string `json:"siblings"`
+	Index        string   `json:"index"`
+	Value        string   `json:"value"`
+	AuxExistence bool     `json:"auxExistence"`
```

Comment:
we should create regular type here
with nodeAux, h_index, h_value.

root is already on upper level. Use merkletree.Proof object

Reply:
ah. maybe it should also contain type property with equality to SparseMerkleTreeProof and other fiellds from mtp

struct {
   type :  ProofType
   merkeTree.Proof
}